auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -823,6 +927,15 @@ private Void replaceIdentityPrivileged(final LoadedIdentity newIdentity) throws
                     } catch (IOException ignored) {
                         // nothing we can do
                     }
+
+                    try {
+                        if (this.publicKey != null) {
+                            createDigitalSignature(this.path.toString(), this.name);
+                            updateMainIndex(this.mainIndex, this.path, this.secretKey);
+                        }
+                    } catch (GeneralSecurityException | IOException e) {
+                        throw ElytronMessages.log.unableToAccessMainIndex(this.path.toString());
```

Comment:
If `createDigitalSignature` throws a `RealmUnavailableException`, we'll get an error about the main index.